Sources/XRPLSwift-rest/Utils/Http.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  http.swift
+//
+//
+//  Created by Bryan on 7/12/24.
+//
+
+import Foundation
+import AnyCodable
+
+typealias APIParams = [String: Any?]
+
+let TimeoutInterval = TimeInterval(3)
+
+open class BaseResponse<T: Codable>: Codable {
+    public var result: T
+    
+    private enum CodingKeys: String, CodingKey {
+        case result = "result"
+    }
+}
+
+open class ErrorResponse: Codable {
+    public var status: String = "error"
+    public var error: String
+    public var errorCode: Int?
+    public var errorMessage: String?
+    public var request: AnyCodable?
+
+    private enum CodingKeys: String, CodingKey {
+        case status = "status"
+        case error = "error"
+        case errorCode = "error_code"
+        case errorMessage = "error_message"
+        case request = "request"
+    }
+}
+
+class http {
+    enum APIError : Error, Equatable {
+        case invalidURL
+        case noData
+        case httpResponseIsNil
+        case ErrorResponse(code: Int?, msg: String?)
+        case unknownError
+    }
+
+    enum HttpMethod: String {
+        case put = "PUT"
+        case post = "POST"
+        case get = "GET"
+        case delete = "DELETE"
+    }
+
+    static func request<T: Codable>(httpMethod : HttpMethod, url: URL, params: APIParams? = nil) async throws -> T {
```

Comment:
Nit. 
Is there a reason why you implemented it separately instead of using a commonly used lib like `Alamofire`?

Reply:
No specific reason but no lib knowledge of general Swift
Plus, about this small http request feature, this implementation would be better than importing that can cause version conflict